server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/MonitorEvent.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.monitor;
+
+import static com.oceanbase.odc.service.monitor.MonitorEvent.MonitorModule.DATASOURCE_MODULE;
+import static com.oceanbase.odc.service.monitor.MonitorEvent.MonitorModule.JOB;
+import static com.oceanbase.odc.service.monitor.MonitorEvent.MonitorModule.SCHEDULE_MODULE;
+
+import org.springframework.context.ApplicationEvent;
+
+import com.oceanbase.odc.service.monitor.datasource.DataSourseMonitorEventHandler;
+import com.oceanbase.odc.service.monitor.datasource.DatasourceMonitorEventContext;
+import com.oceanbase.odc.service.monitor.session.SessionMonitorContext;
+import com.oceanbase.odc.service.monitor.session.SessionMonitorEventHandler;
+import com.oceanbase.odc.service.monitor.task.flow.FlowMonitorEvent;
+import com.oceanbase.odc.service.monitor.task.flow.FlowMonitorEventHandler;
+import com.oceanbase.odc.service.monitor.task.job.TaskJobMonitorEvent;
+import com.oceanbase.odc.service.monitor.task.job.TaskJobMonitorHandler;
+import com.oceanbase.odc.service.monitor.task.schdule.ScheduleMonitorEvent;
+import com.oceanbase.odc.service.monitor.task.schdule.ScheduleMonitorHandler;
+import com.oceanbase.odc.service.schedule.model.ScheduleTaskType;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+@Getter
+public class MonitorEvent<T> extends ApplicationEvent {
```

Comment:
the process construct MonitorEvent with various Context has seen the implementation of Metrics. 
If we want add a new module, we will construct new context, add new handler, modify MonitorHandler to recognize the new context type.
let deliver see MeterHolder will reduce this step.